packages/vital_core/lib/services/utils/json_serializable_converter.dart

```diff
@@ -35,8 +35,9 @@ class JsonSerializableConverter extends JsonConverter {
 
   @override
   Response<ResultType> convertResponse<ResultType, Item>(Response response) {
-    if (ResultType == NoContent || response.statusCode == 204) {
-      return response.copyWith<ResultType>(body: null);
+    if (ResultType == NoContent) {
+      return response.copyWith<NoContent>(body: NoContent())
+          as Response<ResultType>;
```

Comment:
Don't bother parsing the content & just return an instance of `NoContent`.

(Flutter doesn't have a concrete `Void` type)